### PR TITLE
Adds Lettuces. Reworks a number of recipes. Roborgers no longer taste of lettuce. 

### PR DESCRIPTION
**Pull request by Fordoxia**
## What Does This PR Do
Adds lettuce.

Foods that use cabbage when they should be using lettuce now use lettuce (e.g. the BLT).

Burgers that contain lettuce in their sprites now require lettuce to make (except for big bite and super bite burger, which get their lettuce from the initial burger).

Big bite burgers now require a cheese wedge instead of an egg.

Super bite burgers now require a slice of bacon instead of a boiled egg.

The taste of lettuce in roborgers has been replaced with the taste of metal.

## Why It's Good For The Game
Burgers sprites and ingredients now agree with each other.
Cursed substitution of lettuce with cabbage has been removed.
Adds further functional differentiation between sandwiches and burgers.
Roborgers now actually taste of metal, a major ingredient of robot heads.

NPF: Made the recipe datums in the microwave recipes folder all have a uniform format.

## Images of changes
preview

## Testing
Grew and harvested lettuce (and comfrey, which uses the same growth sprites). Also killed some to make sure the dead sprite was there too.

Cooked stuff according to the new recipes, they worked.

Ate roborger, noted the hint of metal that came after the flavour of bun. Could not taste lettuce.

## Changelog
:cl:
add: Adds lettuce,
tweak: Roborger lettuce taste replaced with metal taste. Many burgers now need lettuce. Big and super bite burgers have tweaked recipes. Some other recipe tweeks swapping cabbage with lettuce.
imageadd: lettuce has a unique sprite.
/:cl:

### CODE REVIEW

File: code/modules/food_and_drinks/recipes/recipes_microwave.dm [hunk 1]
```diff
@@ -267,19 +301,27 @@
...
-	items = list(/obj/item/food/snacks/grown/corn)
+	items = list(
+		/obj/item/food/snacks/grown/corn
+	)
 	result = /obj/item/food/snacks/popcorn
 
 /datum/recipe/microwave/spacylibertyduff
 	reagents = list("water" = 5, "vodka" = 5)
-	items = list(/obj/item/food/snacks/grown/mushroom/libertycap, /obj/item/food/snacks/grown/mushroom/libertycap,
-				/obj/item/food/snacks/grown/mushroom/libertycap)
```
Comment (1080pCat) [inline code/modules/food_and_drinks/recipes/recipes_microwave.dm:276]: i think this was the single worst list in the code

File: code/modules/food_and_drinks/recipes/recipes_microwave.dm [hunk 2]
```diff
@@ -1017,19 +1091,25 @@
 
 /datum/recipe/microwave/wrap
 	reagents = list("soysauce" = 10)
-	items = list(/obj/item/food/snacks/friedegg, /obj/item/food/snacks/grown/cabbage )
+	items = list(
+		/obj/item/food/snacks/friedegg,
+		/obj/item/food/snacks/grown/lettuce
+		)
```
Comment (1080pCat) [inline code/modules/food_and_drinks/recipes/recipes_microwave.dm:1097]:
```suggestion
	)
```
github may troll you and make this spaced, but all the others got one indent, but this got two

**Review by 1080pCat — APPROVED** [review]
Lettuces good to me

File: code/modules/food_and_drinks/recipes/recipes_microwave.dm [hunk 3]
```diff
@@ -52,24 +55,49 @@
...
+		/obj/item/food/snacks/grown/lettuce,
 		/obj/item/food/snacks/bun
 	)
 	result = /obj/item/food/snacks/human/burger
 
 /datum/recipe/microwave/plainburger
 	items = list(
 		/obj/item/food/snacks/bun,
+		/obj/item/food/snacks/grown/lettuce,
 		/obj/item/food/snacks/meat //do not place this recipe before /datum/recipe/microwave/humanburger
```
Comment (DGamerL) [inline code/modules/food_and_drinks/recipes/recipes_microwave.dm:67]: While we're here, could you quickly update this comment to say the correct typepath?

File: code/modules/food_and_drinks/recipes/recipes_microwave.dm (hunk 1 above)
Comment (Fordoxia) [reply code/modules/food_and_drinks/recipes/recipes_microwave.dm:276]: It was one of the lists of all time.

File: code/modules/food_and_drinks/recipes/recipes_microwave.dm (hunk 3 above)
Comment (Fordoxia) [reply code/modules/food_and_drinks/recipes/recipes_microwave.dm:67]: Done.

**Comment by Mitchs98** [comment]
While, lettuce probably shouldn't be cabbage in regards to burgers and salads, several other recipes use cabbage. Including some recipes I'll be porting soon. This holds true to TG for a lot of their recipes as well.

I'd rather not make other recipes(including ones I'm slowly working on) make less sense just because there's no lettuce for burgers. I'd personally prefer a lettuce actually be created and added.

In regards to "Lettuce appearing far, far more often in recipes that can be made."; Cabbage is actually the appropriate ingredient for exactly half the recipes that can be created. You typically don't make soup with lettuce nor do you put lettuce into egg wraps/egg rolls or meat buns(unless you're some sorta psychopath anyway).

**Comment by FunnyMan3595** [comment]
Yeah, I'm with Mitchs98 on this one. If we're going to do anything here, I'd rather we make a new plant and split them properl, instead of flipping the problem from "stuff uses cabbage that should use lettuce" to "stuff uses lettuce that should use cabbage".

**Comment by Fordoxia** [comment]
> While, lettuce probably shouldn't be cabbage in regards to burgers and salads, several other recipes use cabbage. Including some recipes I'll be porting soon. This holds true to TG for a lot of their recipes as well.
> 
> I'd rather not make other recipes(including ones I'm slowly working on) make less sense just because there's no lettuce for burgers. I'd personally prefer a lettuce actually be created and added.
> 
> In regards to "Lettuce appearing far, far more often in recipes that can be made."; Cabbage is actually the appropriate ingredient for exactly half the recipes that can be created. You typically don't make soup with lettuce nor do you put lettuce into egg wraps/egg rolls or meat buns(unless you're some sorta psychopath anyway).

This has been addressed. They now coexist.

**Comment by S34NW** [comment]
!merge_upstream

**Comment by S34NW** [comment]
!merge_upstream

**Review by S34NW — APPROVED** [review]

**Pull request merged**